test/e2e/metrics_e2e_test.go

```diff
@@ -149,9 +149,13 @@ var _ = Describe("Metrics are generated for OLM managed resources", func() {
 					restartDeploymentWithLabel(c, "app=olm-operator")
 				})
 				It("CSV metric is preserved", func() {
-					Expect(getMetricsFromPod(c, getPodWithLabel(c, "app=olm-operator"))).To(
-						ContainElement(LikeMetric(WithFamily("csv_succeeded"), WithName(csv.Name), WithValue(1))),
-					)
+					Eventually(func() []Metric {
+						return getMetricsFromPod(c, getPodWithLabel(c, "app=olm-operator"))
+					}).Should(ContainElement(LikeMetric(
```

Comment:
I _think_ this is the right approach when I was looking into this issue. It wasn't clear to me whether we simply were attempting to get the pod metrics before the metrics port was even established, or the startup metric functionality had simply regressed. I'm inclined to think it's the latter, and in certain cluster environments, I could definitely see this test permafailing due to slower container/application start up times.

Reply:
Retrying here is absolutely the correct approach. I suspect this test ran smoothly on the developers machine and does not transfer well to slow ci enviornments as @timflannagan suggested.

Reply:
Thanks for review.!!